src/compiler/crystal/command.cr

```diff
@@ -525,13 +525,24 @@ class Crystal::Command
     first_filename = sources.first.filename
     first_file_ext = File.extname(first_filename)
     original_output_filename = File.basename(first_filename, first_file_ext)
+    {% if flag?(:win32) %}
+      original_output_filename = "#{original_output_filename}.exe"
+    {% end %}
```

Comment:
Shouldn't this technically depend on the target triple, not the host platform? In practice, this doesn't make much difference. But given an appropriate linker is available you could build a win32 executable from a compiler running on a different platform (which shouldn't even be that hard from WSL, I believe).